Sources/CoreMetrics/Metrics.swift

```diff
@@ -165,6 +168,39 @@ public protocol MetricsFactory {
     func makeCounter(label: String, dimensions: [(String, String)]) -> CounterHandler
     func makeRecorder(label: String, dimensions: [(String, String)], aggregate: Bool) -> RecorderHandler
     func makeTimer(label: String, dimensions: [(String, String)]) -> TimerHandler
+
+    /// Signals the `MetricsFactory` that the passed in `MetricHandler` will no longer be updated.
+    /// Implementing this functionality is _optional_, and depends on the semantics of the concrete `MetricsFactory`.
+    ///
+    /// In response to this call, the factory _may_ release resources associated with this metric,
+    /// e.g. in case the metric contains references to "heavy" resources, such as file handles, connections,
+    /// or large in-memory data structures.
+    ///
+    /// **Intended usage:**
+    ///
+    /// Metrics library implementations are _not_ required to act on this signal immediately (or at all).
+    /// However, the presence of this API allows middle-ware libraries wanting to emit metrics for resources with
+    /// well-defined life-cycles to behave pro-actively, and signal when a given metric is known to not be used anymore,
+    /// which can make an positive impact with regards to resource utilization in case of metrics libraries which keep
+    /// references to "heavy" resources.
+    ///
+    /// It is expected that some metrics libraries, may choose to omit implementing this functionality.
+    /// One such example may be a library which directly emits recorded values to some underlying shared storage engine,
+    /// which means that the `MetricHandler` objects themselves are light-weight by nature, and thus no lifecycle
+    /// management and releasing of such metrics handlers is necessary.
+    ///
+    /// **Concurrency:**
+    ///
+    /// This function MAY be invoked concurrently, and implementations should take care to use appropriate
+    /// synchronization mechanisms where necessary.
+    func release<M: MetricHandler>(metric: M)
+}
+
+extension MetricsFactory {
+    public func release<M: MetricHandler>(metric: M) {
+        // no-op by default.
+        // Libraries which do maintain metric lifecycle should implement this method.
+    }
```

Comment:
So implementations which don't care never have to even implement the no-op.

Reply:
I think it might be better to go for explicit denial of lifetime usage, so to say, so that creators of backends are forced to consider if they need to implement this or not

Reply:
Sorry for the radio silence here @MrLotU, back at it now 🙇‍♂️

Either way is likely fine... I think Tom's main concern was to not cause adoption friction for super simple implementations, thus this provided no-op. Maybe others on the SSWG have an opinion here too?

Reply:
/cc @seabaylea